File: utils/src/main/java/com/lab/azure/bootloader/utils/QueryHelper.java
package com.lab.azure.bootloader.utils;

import com.lab.azure.bootloader.utils.arguments.QueryArgument;
import com.microsoft.azure.storage.CloudStorageAccount;
import com.microsoft.azure.storage.StorageException;
import com.microsoft.azure.storage.table.CloudTable;
import com.microsoft.azure.storage.table.CloudTableClient;
import com.microsoft.azure.storage.table.TableQuery;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.io.IOException;
import java.net.URISyntaxException;
import java.security.InvalidKeyException;

public class QueryHelper {
    public static void queryJob(QueryArgument queryArgument) throws URISyntaxException, StorageException, InvalidKeyException, IOException {
        Logger logger = LoggerFactory.getLogger("queryJob");
        logger.info("queryJob {} start", queryArgument.getTask());
        Settings settings = Settings.loadCfg(queryArgument.getSettingFile());
        if (!settings.validate()) {
            logger.warn("settings error");
            System.exit(-1);
        }

        CloudStorageAccount account = CloudStorageAccount.parse(settings.getAzureWebJobsStorage());
        CloudTableClient tableClient = account.createCloudTableClient();
        CloudTable jobHistoryTable = tableClient.getTableReference(settings.getJobHistoryTable());
        jobHistoryTable.createIfNotExists();

        TableQuery<JobEntity> query;

        if("all".equalsIgnoreCase(queryArgument.getTask())){
            query = TableQuery.from(JobEntity.class);
        }
        else{
            String pfilter = TableQuery.generateFilterCondition("PartitionKey", TableQuery.QueryComparisons.EQUAL, queryArgument.getTask());
            String rfilter = TableQuery.generateFilterCondition("RowKey", TableQuery.QueryComparisons.EQUAL, queryArgument.getTask());
            String cfilter = TableQuery.combineFilters(pfilter, TableQuery.Operators.AND, rfilter);
            query = TableQuery.from(JobEntity.class).where(cfilter);
        }

        Iterable<JobEntity> entityIterable = jobHistoryTable.execute(query);
        logger.info("===== show tasks =====");
        for (JobEntity i: entityIterable){
            logger.info("version : {}, task : {}, amount : {}", i.getVersion(), i.getPartitionKey(), i.getAmount());
        }
        logger.info("===== show tasks =====");
        logger.info("queryJob {} stop", queryArgument.getTask());
    }
}

File: src/AzureBootloaderCompiler.Utils/QueryHelper.cs
using Microsoft.WindowsAzure.Storage;
using Microsoft.WindowsAzure.Storage.Table;
using System;
using System.Collections.Generic;
using System.Text;
using System.Threading.Tasks;
using AzureBootloaderCompiler.Utils.Arguments;

namespace AzureBootloaderCompiler.Utils
{
    class QueryHelper
    {
        public static async Task DoQueryAsync(QueryArgument argument)
        {
            Console.WriteLine($"queryJob {argument.Task} start");
            Settings settings = Settings.LoadCfg(argument.SettingFile);
            var account = CloudStorageAccount.Parse(settings.AzureWebJobsStorage);
            var tableClient = account.CreateCloudTableClient();
            var jobHistoryTable = tableClient.GetTableReference(settings.JobHistoryTable);
            _ = jobHistoryTable.CreateIfNotExistsAsync();

            TableQuery<JobEntity> query;

            if ("all".Equals(argument.Task.ToLower()))
            {
                query = new TableQuery<JobEntity>();
            }
            else
            {
                string pfilter = TableQuery.GenerateFilterCondition("PartitionKey", QueryComparisons.Equal, argument.Task);
                string rfilter = TableQuery.GenerateFilterCondition("RowKey", QueryComparisons.Equal, argument.Task);
                string cfilter = TableQuery.CombineFilters(pfilter, TableOperators.And, rfilter);
                query = new TableQuery<JobEntity>().Where(cfilter);
            }

            TableQuerySegment<JobEntity> sengment = await jobHistoryTable.ExecuteQuerySegmentedAsync(query, null);

            List<JobEntity> entities = sengment.Results;
            Console.WriteLine("===== show tasks =====");
            foreach (JobEntity i in entities)
            {
                Console.WriteLine($"version : {i.Version}, task : {i.PartitionKey}, amount : {i.Amount}");
            }
            Console.WriteLine("===== show tasks =====");
            Console.WriteLine($"queryJob {argument.Task} stop");
        }
    }
}
